panel/layout/modal.py

```diff
@@ -0,0 +1,62 @@
+from __future__ import annotations
+
```

Comment:
One thing to consider is if the `Modal` could be made easier to use? Could the api of the `Modal` be generalized to a concept of *actions*/ *action button*.

## Could the api be simplified?

The api of `panel-modal` that I made is:

```python
import panel as pn

from panel_modal import Modal

pn.extension("modal")

modal = Modal(pn.panel("Hi. I am the Panel Modal!", width=200))

pn.Column(modal.param.open, modal).servable()
```

I requires you to include both the button (`modal.param.open`) and the modal `modal`. If you want to customize the button it requires something like `pn.widgets.Button.from_param(modal.param.open`.

Could the api be simplified? Could it be enough to just include the modal?

```python
import panel as pn

from panel_modal import Modal

pn.extension("modal")

pn.Column(
    Modal(pn.panel("Hi. I am the Panel Modal!", width=200))
).servable()
```

That should show the button. And when clicked the modal should open/ close. That would make it much simpler to drop in as you don't need to define a variable holding the `Modal`.

If you want to customize the button you should be able to provide a button or a dictionary of button kwargs.

```python
import panel as pn

from panel_modal import Modal

pn.extension("modal")

pn.Column(
    Modal(
        pn.panel("Hi. I am the Panel Modal!", width=200),
        button=pn.widgets.Button(...., button_style="outline") # could also be = {..., button_style="outline"}
)
).servable()
```

## Could the API be generalized?

A Modal is actually a part of a general concept for *action buttons*. You want to be able to trigger an action via a Button. You want to be able to easily set the content (`object`) and be able to customize the button.

```python
modal_button = ModalButton(object=pn.panel(...), button=...)
copy_button= CopyPasteButton(object="...my code", button=...)
paste_button = PasteButton(object=pn.widgets.Tabulator, button=...., ) # Here the object is the target. Not the source.
link_button = LinkButton(object="https://panel.holoviz.org, target="_blank", button=...)

maximize_content = pn.Column(...)
maximize_button = MaximizeButton(object=maximize_content, button=...) # Maximizes the object to full window size
...
```

## Alternative Api

Add a method - for example `.layout` to include then `object` if relevant.

```python
import panel as pn

from panel_modal import Modal

pn.extension("modal")

pn.Column(
    Modal(
        pn.panel("Hi. I am the Panel Modal!", width=200),
        button=pn.widgets.Button(...., button_style="outline") # could also be = {..., button_style="outline"}
).layout()
).servable()
```

@philippjfr. What do you think?